creator-node/src/services/stateMachineManager/stateReconciliation/updateReplicaSet.jobProcessor.ts

```diff
@@ -288,17 +288,19 @@ const _determineNewReplicaSet = async ({
 }: DetermineNewReplicaSetParams) => {
   const currentReplicaSet = [primary, secondary1, secondary2]
   const healthyReplicaSet = new Set(
-    currentReplicaSet.filter((node) => !unhealthyReplicasSet.has(node))
+    currentReplicaSet.filter((node) => node && !unhealthyReplicasSet.has(node))
   )
+  const numberOfEmptyReplicas = currentReplicaSet.filter((node) => !node).length
```

Comment:
we previously overlooked replacing empty nodes. since an empty string isn't unique, it doesn't make sense to add it to a set. thus, `numberOfEmptyReplicas` will be replaced in addition to the node(s) that the findReplicaSetUpdates job marked unhealthy